src/main/kotlin/org/jetbrains/intellij/dependency/PluginsRepositoryConfiguration.kt

```diff
@@ -20,6 +21,13 @@ abstract class PluginsRepositoryConfiguration {
         pluginsRepositories.add(MavenPluginsRepository(url))
     }
 
+    /**
+     * Use a Maven repository closure
+     */
+    fun maven(mavenRepo: Closure<Any>) {
```

Comment:
Will it work in `build.gradle.kts`? I'm not aware whether kts can seamlessly work with groovy closures

Reply:
Please define the exact input structure. No `Any` will be accepted anymore.

Reply:
@zolotov added support for build.gradle.kts

Reply:
@hsz I don't have any idea what to set except as `Any`.

The [official documentation](https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/dsl/RepositoryHandler.html#maven-groovy.lang.Closure-) doesn't give answer. If you know the exact input structure tell please, I will be fixing it

Reply:
`Action<T>` is now preferred for handling such a functionality over Groovy-specific `Closure`.
Try to set breakpoints in both methods: `maven(Closure)` and `maven(Action)` – at first, Groovy-based configuration will stop in the first one, but as soon as you'll remove it from the code base, it'll pick the second method anyway.
So for me, `MavenRepositoryPluginByClosure` is not needed at all.

BTW – could you please provide tests for this way of Maven configuration?

Reply:
@hsz 
`MavenRepositoryPluginByClosure` delete it.

Reply:
@hsz I added test for that maven configuration. But I can't understand why the test `reuse configuration cache` from BuildSearchableOptionsTaskSpec failed. I have checked it on local machine all worked fine. Could you help me?

Reply:
@artsok Looks like failed tests were caused by some odd weather conditions around GitHub HQ because further runs failed because of other causes, like networking.
Everything looks good now, so if you're done – we could merge and release it within the upcoming 1.0.1.

Reply:
@hsz yep, I finished with that PR. You can merge it. Thx!